LibraryInstrumentationGuidance.md

```diff
@@ -1,53 +1,115 @@
 
-## Instrumentation 
+## Bare Minimum Instrumentation 
 
-In the simplest case, the operation that is being monitored has to be wrapped by an activity. However in order to minimize performance impact the activity should be only created if there is any listener waiting for it.
+In the simplest case, the operation that is being monitored has to be wrapped by an activity. 
 
 ```csharp
+    // create DiagnosticListener with unique name
     static DiagnosticListener source = new DiagnosticListener("Example.MyLibrary");
 
     Activity activity = null;
-    // create activity only when requested
-    if (source.IsEnabled() && source.IsEnabled("Example.MyLibrary.MyOperation"))
+    
+    // check if listener is interested in this operation
+    if (source.IsEnabled() && source.IsEnabled("MyOperation"))
     {
-        activity = new Activity("Example.MyLibrary.MyOperation");
-        source.StartActivity(activity, new { Input = input });
+        activity = new Activity("MyOperation");
+        activity.Start();
     }
 
     object output = null;
     try
     {
         // perform the actual operation
-        output = RunOperation(input);  
+        output = RunOperation(input);
+    }
+    catch (Exception ex)
+    {
+        activity?.AddTag("error", true);
     }
     finally
     {
         // stop activity if started
         if (activity != null)
-             source.StopActivity(activity, new { Input = input, Output = output }); 
+             source.StopActivity(activity, new MyStopPayload { Input = input, Output = output }); 
     }
 ```
 > ### *__TODO__ - provide a pointer to a document with more advanced instrumentation examples (WIP)* 
 
+This minimum instrumentation give tracing system a chance to trace something like '<timestamp> [Example.MyLibrary] MyOperation, Id 'abc.1.2.3.', took <duration> ms, with properties [{ "error": true }]'.
+While in majority of cases some other details are desirable (and we will get to it [here](TBD)), this instrumentation gives bare minimum.
+
+### Instrumentation control and sampling considerations
+
+**DO** make `DiagnosticListener` name globally unique.
+Every DiagnosticListener has a name. Tracing system discovers all available sources and subscribes to them based on the name. 
+[DiagnosticSource guide](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/DiagnosticSourceUsersGuide.md#naming-conventions) provides more naming considerations.
```

Comment:
As I've said in #913, this document should be merged with the existing UserGuides.